MovingBlocks/Terasology
# Java 18 - do not set SecurityManager if unsupported

### Contains
This pull request reverts https://github.com/MovingBlocks/Terasology/commit/ca913261d6564ac552b067a234b6418c2a3a9ec7 and re-implements it properly using a runtime check. It ensures that the game does not set a custom `SecurityManager` when running on Java 18 or above.

### How to test
- Run the game using Java 18 (or later)
- Ensure that the game doesn't crash on start-up (it should reach at least the main menu)
